video/vdu_audio.h

```diff
@@ -9,34 +9,35 @@
 
 #include <memory>
 #include <vector>
-#include <array>
+#include <unordered_map>
 #include <fabgl.h>
 
 fabgl::SoundGenerator		SoundGenerator;		// The audio class
 
 #include "agon.h"
 #include "agon_audio.h"
 #include "vdp_protocol.h"
+#include "types.h"
 
 // audio channels and their associated tasks
-std::array<std::shared_ptr<audio_channel>, MAX_AUDIO_CHANNELS> audio_channels;
-std::vector<TaskHandle_t> audioHandlers;
+std::unordered_map<uint8_t, std::shared_ptr<audio_channel>> audio_channels;
+std::vector<TaskHandle_t, psram_allocator<TaskHandle_t>> audioHandlers;
 
-std::array<std::shared_ptr<audio_sample>, MAX_AUDIO_SAMPLES> samples;	// Storage for the sample data
+std::unordered_map<uint8_t, std::shared_ptr<audio_sample>> samples;	// Storage for the sample data
```

Comment:
using `std::unordered_map` for these things means that we're not just grabbing a chunk of internal RAM in one hit.  it's less efficient if the user enables lots of channels and sound samples, but for typical usage will result in a fairly significantly smaller amount of RAM being used